src/lib/io/sol-iio.c

```diff
@@ -700,12 +741,33 @@ sol_iio_open(int device_id, const struct sol_iio_config *config)
         return NULL;
     }
 #endif
+    SOL_LIST_FOREACH (&iio_open_list, it) {
+        entry = SOL_LIST_GET_CONTAINER(it, struct iio_opened_device, list);
+        if (device_id == entry->device->device_id) {
+            if ((config->buffer_size == entry->config->buffer_size)
+                && (config->sol_iio_reader_cb == entry->config->sol_iio_reader_cb)
```

Comment:
Different 'readers callbacks' should be allowed, don't you think? If a device has both gyro and temperature, why limiting to the very same callback?